File: app/src/main/java/org/imozerov/streetartview/AppComponent.java
package org.imozerov.streetartview;

import dagger.Component;

import org.imozerov.streetartview.ui.detail.DetailArtObjectActivity;
import org.imozerov.streetartview.ui.detail.ImageViewActivity;
import org.imozerov.streetartview.ui.explore.all.ArtListFragment;
import org.imozerov.streetartview.ui.explore.base.ArtListPresenter;
import org.imozerov.streetartview.ui.explore.favourites.FavouritesListFragment;
import org.imozerov.streetartview.ui.explore.map.ArtMapFragment;

import javax.inject.Singleton;

/**
 * Created by imozerov on 06.02.16.
 */
@Singleton
@Component(modules = AppModule.class)
public interface AppComponent {
    void inject(ArtListPresenter artListPresenter);
    void inject(DetailArtObjectActivity detailArtObjectActivity);
    void inject(ImageViewActivity imageViewActivity);


    void inject(ArtMapFragment artMapFragment);
    void inject(ArtListFragment artListFragment);
    void inject(FavouritesListFragment favouritesListFragment);
}

File: app/src/main/java/org/imozerov/streetartview/AppComponent.kt
package org.imozerov.streetartview

import dagger.Component
import org.imozerov.streetartview.location.LocationService
import org.imozerov.streetartview.network.FetchService
import org.imozerov.streetartview.network.NetModule
import org.imozerov.streetartview.ui.add.AddInfoActivity
import org.imozerov.streetartview.ui.add.PickImageActivity
import org.imozerov.streetartview.ui.detail.DetailArtObjectActivity
import org.imozerov.streetartview.ui.detail.DetailArtObjectFragment
import org.imozerov.streetartview.ui.detail.ImageViewActivity
import org.imozerov.streetartview.ui.explore.ExploreArtActivity
import org.imozerov.streetartview.ui.explore.base.ArtListPresenter
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(AppModule::class, NetModule::class))
interface AppComponent {
    fun inject(locationService: LocationService)
    fun inject(fetchService: FetchService)
    fun inject(artListPresenter: ArtListPresenter)
    fun inject(exploreArtActivity: ExploreArtActivity)
    fun inject(pickImageActivity: PickImageActivity)
    fun inject(addInfoActivity: AddInfoActivity)
    fun inject(detailArtObjectActivity: DetailArtObjectActivity)
    fun inject(detailArtObjectFragment: DetailArtObjectFragment)
    fun inject(imageViewActivity: ImageViewActivity)
}
